core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -734,9 +734,20 @@ public BytesReference ext() {
         return ext;
     }
 
+    /**
+     * Create a new SearchSourceBuilder with attributes set by an xContent.
+     */
     public SearchSourceBuilder fromXContent(XContentParser parser, QueryParseContext context, AggregatorParsers aggParsers)
             throws IOException {
         SearchSourceBuilder builder = new SearchSourceBuilder();
+        builder.parseXContent(parser, context, aggParsers);
```

Comment:
It's because the search request that reindex uses has different defaults than the regular search request. This lets it push the defaults into the search request before the parsing happens. Should I do this in another way? Add a comment to parseXContent about it?